### PR TITLE
Refactor config panel

### LINKED ISSUE #1236: Refactor config panels

### PR DESCRIPTION
**Pull request by holderb5**
Resolves #1236 

## Description
Refactor the config panels for visualizations to use denali stylings

-

## Screenshots
Screen Shot 2021-02-11 at 3 10 21 PM
Screen Shot 2021-02-11 at 1 55 33 PM
Screen Shot 2021-02-11 at 1 14 50 PM
Screen Shot 2021-02-11 at 1 14 34 PM
Screen Shot 2021-02-11 at 1 10 06 PM
Screen Shot 2021-02-11 at 3 13 44 PM

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/tests/integration/components/navi-visualization-config/line-chart-test.ts [hunk 1]
```diff
@@ -85,8 +83,8 @@ module('Integration | Component | visualization config/line chart', function(hoo
       .dom('.line-chart-config .line-chart-config__series-config')
       .isVisible('The series config component is rendered when `stacked` option is on');
 
-    await click('.line-chart-config__stacked-opt');
-    assert.dom('.line-chart-config__stacked-opt').isNotChecked('Stacked it not true');
+    await click('.line-chart-config__stacked-opt-select');
+    assert.dom('.line-chart-config__series-config').isVisible('Stacked it not true');
```
Comment (kevinhinterlong) [inline packages/core/tests/integration/components/navi-visualization-config/line-chart-test.ts:87]: Why did this change? Also can we update the assert message

File: packages/core/addon/components/navi-visualization-config/series-chart.hbs [hunk 2]
```diff
@@ -2,8 +2,7 @@
 <div class="series-chart-config" ...attributes>
   {{#if this.showMetricSelect}}
     <div class="metric-select">
-      <div class="metric-select__header">Metric</div>
-      <div class="metric-select__select">
+      <DenaliInputGroup @label="Metric" class="metric-select__select m-b-30">
         <PowerSelect
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/series-chart.hbs:6]: Can we make this full width?

File: packages/core/addon/components/navi-visualization-config/goal-gauge.hbs [hunk 3]
```diff
@@ -19,7 +19,7 @@
   <DenaliInputGroup
     @label="Baseline"
     @isStacked= {{true}}
-    class="goal-gauge-config__section-header"
+    class="goal-gauge-config__section-header m-b-30"
   >
     <DenaliInput
       @size="medium"
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/goal-gauge.hbs:25]:
```suggestion
      @size="medium"
      step="any"
```

File: packages/core/addon/components/navi-visualization-config/goal-gauge.hbs [hunk 4]
```diff
@@ -2,9 +2,9 @@
...
 
    <DenaliInputGroup
-     @label="Label"
+     @label="Goal Label"
      @isStacked={{true}}
-     class="goal-gauge-config__section-header"
+     class="goal-gauge-config__section-header m-b-30"
    >
     <DenaliInput
       @size="medium"
```
Comment (kevinhinterlong) [inline packages/core/addon/components/navi-visualization-config/goal-gauge.hbs:10]: This should fix the red bar that shows up by default
```suggestion
      @size="medium"
      step="any"
```

File: packages/core/tests/integration/components/navi-visualization-config/line-chart-test.ts (hunk 1 above)
Comment (holderb5) [reply packages/core/tests/integration/components/navi-visualization-config/line-chart-test.ts:87]: would having it be stacked-opt-toggle make more sense than select? think the class name just changed slightly. As far as the test goes, the animation wasnt triggering in time for the assert to see that the toggle is actually checked. I tried to add in an await animationsSettled, but the import file was unfound

File: packages/core/addon/components/navi-visualization-config/goal-gauge.hbs (hunk 3 above)
Comment (holderb5) [reply packages/core/addon/components/navi-visualization-config/goal-gauge.hbs:25]: yup that did it

File: packages/core/app/styles/navi-core/components/navi-visualization-config/series-chart.less [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/**	
+ * Copyright 2019, Yahoo Holdings Inc.	
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.	
+ */
+
+.metric-select {
+  &__select-trigger {
+    flex: 1;
+  }
```
Comment (kevinhinterlong) [inline packages/core/app/styles/navi-core/components/navi-visualization-config/series-chart.less]: Going forward we should use `.scss` files and add `index.scss` and imports as needed up to the root `.scss` file, but we can probably just remove this style and use the `w-full` helper for the `@triggerClass` of the power select

Comment (holderb5) [reply packages/core/app/styles/navi-core/components/navi-visualization-config/series-chart.less]: Oh you're right I forgot that denali has helpers for flex box as well, I'll re-delete the file and add the helper on instead.

File: packages/core/app/styles/navi-core/components/subtotal-dimension-select-trigger.less [hunk 6]
```diff
@@ -4,6 +4,8 @@
 */
 
 .subtotal-dimension-select-trigger {
+  width: 100%;
```
Comment (jkusa) [inline packages/core/app/styles/navi-core/components/subtotal-dimension-select-trigger.less:7]: replace with `w-full`?

File: packages/core/app/styles/navi-core/components/subtotal-dimension-select-trigger.less [hunk 7]
```diff
@@ -4,6 +4,8 @@
...
+
```
Comment (jkusa) [inline packages/core/app/styles/navi-core/components/subtotal-dimension-select-trigger.less:8]: Do we still have/need `&__prefix` ?

File: packages/core/app/styles/navi-core/utils/mixins.less [hunk 8]
```diff
@@ -74,19 +74,13 @@
 }
 
 .config-container {
```
Comment (jkusa) [inline packages/core/app/styles/navi-core/utils/mixins.less:76]: Could this be removed?

File: packages/core/app/styles/navi-core/utils/mixins.less [hunk 9]
```diff
@@ -74,19 +74,13 @@
...
-  background-color: @navi-gray-100;
-  font-family: @font-family-thin;
+  background-color: @navi-white;
   margin: 5px;
   padding: 10px;
   text-align: left;
 }
 
 .config-section {
```
Comment (jkusa) [inline packages/core/app/styles/navi-core/utils/mixins.less:83]: Could this be cleaned up?

File: packages/core/app/styles/navi-core/components/subtotal-dimension-select-trigger.less (hunk 7 above)
Comment (holderb5) [reply packages/core/app/styles/navi-core/components/subtotal-dimension-select-trigger.less:8]: I could be wrong but I dont think we need this entire file anymore

File: packages/core/addon/components/chart-series-config.hbs [hunk 10]
```diff
@@ -1,9 +1,8 @@
 {{!-- Copyright 2020, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
 <div class="line-chart-config__series-config" ...attributes>
-  <div class="line-chart-config__series-config__header">Stack Order</div>
-  <div class="line-chart-config__series-config__content">
+  <DenaliInputGroup @label="Stack Order" @isStacked={{true}}>
     <SortableGroup
-      class="line-chart-config__series-config__collection"
+      class="line-chart-config__series-config__collection m-t-10"
```
Comment (jkusa) [inline packages/core/addon/components/chart-series-config.hbs:5]:
```suggestion
      class="line-chart-config__series-config-collection m-t-10"
```

File: packages/core/app/styles/navi-core/components/navi-visualization-config/goal-gauge.less [hunk 11]
```diff
@@ -4,9 +4,5 @@
  */
 
 .goal-gauge-config {
-  .config-container();
   align-items: center;
```
Comment (jkusa) [inline packages/core/app/styles/navi-core/components/navi-visualization-config/goal-gauge.less:7]: Required?

File: packages/core/app/styles/navi-core/components/navi-visualization-config/line-chart.less [hunk 12]
```diff
@@ -4,37 +4,8 @@
  */
 
 .line-chart-config {
-  .config-container();
   padding: 5px;
```
Comment (jkusa) [inline packages/core/app/styles/navi-core/components/navi-visualization-config/line-chart.less:7]: Is this still needed?

File: packages/core/app/styles/navi-core/components/navi-visualization-config/line-chart.less [hunk 13]
```diff
@@ -4,37 +4,8 @@
...
-    }
-  }
-
-  &__area-opt,
-  &__stacked-opt {
-    &-select {
-      padding-right: 8px;
-    }
-  }
-
```
Comment (jkusa) [inline packages/core/app/styles/navi-core/components/navi-visualization-config/line-chart.less:37]: Styles below still needed for `__curve-opt`?

File: packages/core/app/styles/navi-core/components/number-format-selector.less [hunk 14]
```diff
@@ -16,8 +16,4 @@
   &__format-input {
     font-size: @font-size-base-large;
   }
-
```
Comment (jkusa) [inline packages/core/app/styles/navi-core/components/number-format-selector.less:19]: do we need `&__doc-link` & `&__format-input` styles?

File: packages/core/tests/acceptance/line-chart-test.ts [hunk 15]
```diff
@@ -79,14 +79,11 @@ module('Acceptance | line chart', function(hooks) {
     await visit('/line-chart');
 
     // switch on `stacked` and expand the config
-    await click('.chart-container.metric .line-chart-config__stacked-opt .denali-switch');
-    await click('.line-chart-config__series-config__header');
+    await click('.chart-container.metric .line-chart-config__stacked-opt-select');
```
Comment (jkusa) [inline packages/core/tests/acceptance/line-chart-test.ts:82]:
```suggestion
    await click('.line-chart-config__stacked-opt-select');
```
does this work?

File: packages/core/tests/acceptance/line-chart-test.ts [hunk 16]
```diff
@@ -124,8 +121,7 @@ module('Acceptance | line chart', function(hooks) {
     await visit('/line-chart');
 
     // switch on `stacked` and expand the config
-    await click('.chart-container.dimension .line-chart-config__stacked-opt .denali-switch');
-    await click('.chart-container.dimension .line-chart-config__series-config__header');
+    await click('.chart-container.dimension .line-chart-config__stacked-opt-select');
```
Comment (jkusa) [inline packages/core/tests/acceptance/line-chart-test.ts:124]:
```suggestion
    await click('.line-chart-config__stacked-opt-select');
```

File: packages/reports/app/styles/navi-reports/components/report-view.less [hunk 17]
```diff
@@ -40,8 +40,8 @@
   }
 
   &__visualization-edit {
-    background-color: @navi-modal-gray;
-    border-left: 1px solid @navi-lighter-gray;
+    background-color: @navi-white;
```
Comment (jkusa) [inline packages/reports/app/styles/navi-reports/components/report-view.less:43]:
```suggestion
    background-color: @denali-grey-100;
```

File: packages/core/tests/acceptance/line-chart-test.ts (hunk 16 above)
Comment (holderb5) [reply packages/core/tests/acceptance/line-chart-test.ts:124]: we need these. Its on a page with many line charts so the .dimension differentiates between them

Comment (jkusa) [reply packages/core/tests/acceptance/line-chart-test.ts:124]: ah, makes sense 👍

File: packages/core/app/styles/navi-core/components/navi-visualization-config/line-chart.less (hunk 13 above)
Comment (holderb5) [reply packages/core/app/styles/navi-core/components/navi-visualization-config/line-chart.less:37]: Those are for the stacked option, for displaying all of the columns, I'll delete a few and see what happens

Comment (holderb5) [reply packages/core/app/styles/navi-core/components/navi-visualization-config/line-chart.less:37]: Those are all necessary for the stack order styling

**Review by kevinhinterlong — APPROVED** [review]

**Review by jkusa — APPROVED** [review]

**Pull request merged**